### ISSUE TITLE
Input lag

### ISSUE THREAD

**Issue by BrundleThwaite** [opentoonz/opentoonz#1592]
I am getting a delay of about a second when drawing. This happens on all the releases after 2017-10-26 09:31:23 UTC nightly_fd55f636ceffa26207b6ecedee5516a2fd45d77e

It happens with both my cintiqe hd 13 and my wacom intuos 4 on both my laptop and my desktop.

**Comment by artisteacher** [comment]
Does it happen with a mouse too?

**Comment by BrundleThwaite** [comment]
It is very difficult to draw with a mouse and I haven't used one for years at this point I don't own a mouse.

**Comment by BrundleThwaite** [comment]
![lag0](https://user-images.githubusercontent.com/18170954/33419022-49225f4c-d5b1-11e7-8487-f84b6ca50703.gif)

This should give you some idea of the lag. At the worst I have had up to a centimeter of delay between the cursor and the drawn line.

**Comment by turtletooth** [comment]
@shun-iwasawa There are a number of lag reports on the discord server.  Do you think Qt 5.9 is causing the lag?

@jpturcotte posted this as the most recent version without the lag, but this is after the qt change, right?
https://github.com/opentoonz/opentoonz_nightlies/releases/tag/nightly_79662fa9d23c8f833d7a033ed8968ac8526d03f5

**Comment by BrundleThwaite** [comment]
In my case the lag starts on this version.
[here](https://github.com/opentoonz/opentoonz_nightlies/releases/tag/nightly_4183496bdcae493542012d4c1a8ca67b89e3328a)

It is the version with the switch to QOpenGL I have check all the releases before and after this when I placed the original post.

In my case the latest working version is this one (It is the version I am using without lag).
[Version I am using](https://github.com/opentoonz/opentoonz_nightlies/releases/tag/nightly_fd55f636ceffa26207b6ecedee5516a2fd45d77e)

**Comment by shun-iwasawa** [comment]
@turtletooth 

> @shun-iwasawa There are a number of lag reports on the discord server. Do you think Qt 5.9 is causing the lag?

Yes, Qt 5.9 + QOpenGLWidget seem to be a possible cause of the reported lag, since the newer version of toolkit is usually more resource intensive.

@BrundleThwaite 
Can you please let me know the information of your CPU, GPU, RAM and file system storage?
Also please make sure that the GPU driver is updated.

**Comment by BrundleThwaite** [comment]
OS: Microsoft Windows 10 Pro
Version: 10.0.16299 Build 16299.
Processor: AMD FX(tm)-8300 Eight-Core Processor, 3300 Mhz, 4 Core(s), 8 Logical Processor(s)
Installed Physical Memory (RAM)	8,00 GB
Drive 1: 232.40 GB SSD
Drive 1: 4TB Seagate

Graphics card:  Radeon(TM) RX 460 Graphics 2G

OpenToonz is running off the SSD.

I am up to date on my drivers.

**Comment by shun-iwasawa** [comment]
@BrundleThwaite 

Thank you for the info. Hmm, It seems to be better than my Win10 MobileStudioPro with the following specs:
```
OS: Win10Pro (10.0.14393)
Processor: Intel i7-6567U
RAM: 8GB
Drive: 256GB SSD
Graphics: Intel Iris Graphics 550
```

I couldn't find such a notable lag on that.
The difference is that you are on AMD & Radeon, but I have none of them for further verification.

As @artisteacher previously mentioned, I also would like to know the lag occurs with a mouse or not.

**Comment by BrundleThwaite** [comment]
Sorry I don't have a mouse. 

I will try to check on my laptop as well If I get the time.

**Comment by manongjohn** [comment]
@BrundleThwaite can ypu verify if OT is using GPU? If it isn't, try forcing it in your graphics card setting.

**Comment by jpturcotte** [comment]
> @jpturcotte posted this as the most recent version without the lag, but this is after the qt change, right?
> https://github.com/opentoonz/opentoonz_nightlies/releases/tag/nightly_79662fa9d23c8f833d7a033ed8968ac8526d03f5

I was going from a post by @BrundleThwaite, but it seems I was a couple builds off.

**Comment by BrundleThwaite** [comment]
@manongjohn  Forced OT to use my GPU (via the Radeon settings) But there was no change.

**Comment by jpturcotte** [comment]
Grimik confirmed on the Discord server that the November nightly I mistakenly posted also lags, but that the October one @BrundleThwaite posted is lag-free. Here is a comparison shot Grimik made: 
![comparasion](https://user-images.githubusercontent.com/26857126/33960294-4a20b5d0-e018-11e7-84a5-0ebaa4a441e6.jpg)

**Comment by turtletooth** [comment]
@shun-iwasawa This one has been leaving me confused about what to do.  I think it's the introduction of either qopenglwidget or qt 5.9 that did this.  Introducing qt 5.9 solved a ton of problems for mac, but it may have hurt the drawing experience on lower end hardware.  

What are your thoughts on this?

**Comment by shun-iwasawa** [comment]
@turtletooth 
IMO it would better to optimize the source based on Qt5.9. 
We are on halfway of transition to the "modern" OpenGL and for now both the "conventional" and the "modern" OpenGL code are mixed up in the source. Such situation may be causing some slowness.

**Comment by turtletooth** [comment]
That makes sense.  I have just seen a handful of comments regarding lag from other people and was wondering your thoughts.

**Comment by jpturcotte** [comment]
We risk losing a lot of people with this update. The effect has already been felt on Discord and Twitter. Is there anything we can do to help speed up that transition? Also, as @snapai proposed on Discord, could we establish a release candidate system for the next release? Thank you!

**Comment by RodneyBaker** [comment]
Isn't every nightly release a release candidate?

**Comment by jpturcotte** [comment]
@RodneyBaker I do not think many people use or even know of the nightlies. I know it is a weak proof, but if common users made use of the nightly releases, this issue would have been caught in November. One could argue it should have been caught anyway before the official release. I believe RCs would have helped.

**Comment by RodneyBaker** [comment]
@jpturcotte I think your suggestion to have a few Release Candidates is a good one.
You should start a new Issue and @shun-iwasawa can make a decision on that..
All that would need to be done is to duplicate a nightly at a specific moment and tag it as a Release Candidate.  

You raise a valid point.  I'm not trying to diminish that.
The vast majority of users don't want to test software... and of those a majority may struggle regardless... even if all the features work and there are no issues to contend with related to hardware variables. 

As for Nightlies
That's our (collective) fault for not getting the word out about the nightlies and the importance (especially with open source software) of the feedback loop involved.  The first step of troubleshooting any issue should be to install the latest nightly build.  While that could very well break something (in fact that is very likely) it is also the best bet going toward resolving known problems.
 
Education-wise
I just read on Twitter where someone was stating OpenToonz was too much for them and they wanted recommendations for other animation software.  Someone needs to befriend that animation wannabe, take them under their wing and get them up to speed with OpenToonz!  That's why we have a community right?  Of course the issue you raise directly effects/impacts developers who have to be on guard and ready especially when a new release is immanent... because unforseen problems will be encountered!  And that leads us back to you point...

Release Candidates
Sounds like a good idea.  Now we just have to get @shun-iwasawa on board with it.

Bottom line:  This may be largely a matter of wording/titling.  If a November nightly release had been duplicated and tagged as v1.2 RC1 that certainly would have signaled to the world that breaking changes were on the horizon.  I'm a bit skeptical that it would have met with significantly different results than what you have described.

**Comment by turtletooth** [comment]
@BrundleThwaite Could you please try the modified build I posted on gumroad today?  It has all the new features of 1.2, except the ability to use shader fx, but I'm hoping it restores the pen behavior.

**Comment by snapai** [comment]
Not at all. Nightlies are usually testing builds of the main branch, typically automated, to see how new PRs are affecting the build, and assumed to be unstable.
Release Candidate builds however, are from the feature-freeze branch for a specific release, and are issued manually. They're promoted in announcements for wider beta testing as ,"this could be the official release for the next version, please help us find any remaining issues with it before the release" - typically projects will have 1-3 RC's issued before the official release, depending on how many bugs people find.


Sent from my Sprint Samsung Galaxy Note5.
-------- Original message --------From: Rodney <notifications@github.com> Date: 12/14/17  8:25 AM  (GMT-05:00) To: opentoonz/opentoonz <opentoonz@noreply.github.com> Cc: snapai <snapai@weanimatethings.com>, Mention <mention@noreply.github.com> Subject: ***UNCHECKED*** Re: [opentoonz/opentoonz] Input lag in nightly builds
 	(#1592) 
Isn't every nightly release a release candidate?



On Thu, Dec 14, 2017 at 7:20 AM, J.P. Turcotte <notifications@github.com>

wrote:

**Comment by BrundleThwaite** [comment]
I Just wan to update that I still have lag issues on the newest 1.2.1 build. But I do want to add that there is no lag when I use the locator for vector drawings and very little when I use raster brushes. 

Using the locator to draw with It is unfortunately not a great solution to the problem though.

**Comment by artisteacher** [comment]
Is the lag mostly for vector levels then? Has it improved at all?

**Comment by BrundleThwaite** [comment]
The lag is on all levels and has not really improved on any of the updates. I was surprised to see that It is not present in the inspector. I never really use the inspector it was an accident that I even drew in it.

**Comment by turtletooth** [comment]
Closing.  Discussion at #1638
